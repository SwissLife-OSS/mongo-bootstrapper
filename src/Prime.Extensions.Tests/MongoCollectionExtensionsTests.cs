using System;
using System.Collections.Generic;
using System.Text;
using MongoDB.Driver;
using Snapshooter.Xunit;
using Squadron;
using System.Threading.Tasks;
using Xunit;

namespace MongoDB.Prime.Extensions.Tests
{
    public class MongoCollectionExtensionsTests : IClassFixture<MongoResource>
    {
        private readonly IMongoDatabase _mongoDatabase;

        public MongoCollectionExtensionsTests(MongoResource mongoResource)
        {
            _mongoDatabase = mongoResource.CreateDatabase();
        }

        [Fact(Skip = "not ready yet.")]
        public async Task Explain_ExplainSingleQuery_SuccessfullyExplained()
        {
            // Arrange
            IMongoCollection<Bar> barCollection =
                _mongoDatabase.GetCollection<Bar>();

            FindOptions findOptions = new FindOptions
            {
                Collation = new Collation(
                    locale: "en",
                    strength: CollationStrength.Secondary)
            };

            FilterDefinition<Bar> filter =
                Builders<Bar>.Filter.And(
                    Builders<Bar>.Filter.Eq(u => u.Id, "1234"),
                    Builders<Bar>.Filter.Eq(b => b.Name, "NN"));

            // Act
            string? mongodbExplain = barCollection.Explain(filter, findOptions);            

            // Assert
            Snapshot.Match(mongodbExplain);
        }
    }
}
