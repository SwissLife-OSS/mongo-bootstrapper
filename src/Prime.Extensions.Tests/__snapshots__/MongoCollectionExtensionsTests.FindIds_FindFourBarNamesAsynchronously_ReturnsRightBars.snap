[
  {
    "Key": "Bar3",
    "Value": {
      "Id": "a1c9e3e8-b448-42da-a684-716932903043",
      "Name": "Bar3",
      "Value": "Value3"
    }
  },
  {
    "Key": "Bar4",
    "Value": {
      "Id": "a1c9e3e8-b448-42da-a684-716932903044",
      "Name": "Bar4",
      "Value": "Value4"
    }
  },
  {
    "Key": "Bar5",
    "Value": {
      "Id": "a1c9e3e8-b448-42da-a684-716932903045",
      "Name": "Bar5",
      "Value": "Value5"
    }
  },
  {
    "Key": "Bar6",
    "Value": {
      "Id": "a1c9e3e8-b448-42da-a684-716932903046",
      "Name": "Bar6",
      "Value": "Value6"
    }
  }
]
